### PR TITLE
Disable node eviction during cluster deletion

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
The machine-controller introduces node draining in newer versions unless a special annotation is set on the node.
This PR will set the required annotation on all nodes to disable the node eviction during cluster deletion.

Also the node deletion was removed as it was for backwards compatibility for the node-controller - which was replaced with the machine-controller before kubermatic v2.5 (Which is the oldest deployed version)

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/controller/cluster/deleting.go [hunk 1]
```diff
@@ -36,48 +36,54 @@ func (cc *Controller) deletingNodeCleanup(c *kubermaticv1.Cluster) (*kubermaticv
 		return c, nil
 	}
 
-	machineClient, err := cc.userClusterConnProvider.GetMachineClient(c)
+	clusterClient, err := cc.userClusterConnProvider.GetClient(c)
```
Comment (alvaroaleman) [inline api/pkg/controller/cluster/deleting.go]: To me this looks like it was a client for the `Cluster.k8s.io` APIGroup ;) Can we rename it to `userClusterCoreClient` or something the like?

File: api/pkg/controller/cluster/deleting.go [hunk 2]
```diff
@@ -36,48 +36,54 @@ func (cc *Controller) deletingNodeCleanup(c *kubermaticv1.Cluster) (*kubermaticv
...
+	// If we delete a cluster, we should disable the eviction on the nodes
+	for _, node := range nodes.Items {
+		// TODO: Use const from machine-controller repo when updated
+		const skipEvictionAnnotation = "kubermatic.io/skip-eviction"
+		if node.Annotations[skipEvictionAnnotation] != "true" {
+			if node.Annotations == nil {
+				node.Annotations = map[string]string{}
+			}
+			node.Annotations[skipEvictionAnnotation] = "true"
+			if _, err := clusterClient.CoreV1().Nodes().Update(&node); err != nil {
```
Comment (alvaroaleman) [inline api/pkg/controller/cluster/deleting.go]: This has a pretty decent chance to fail as nodes get updated very often, can we use some wrapper func that uses `retry.RetryOnConflict`?

File: api/pkg/controller/cluster/deleting.go [hunk 3]
```diff
@@ -36,48 +36,54 @@ func (cc *Controller) deletingNodeCleanup(c *kubermaticv1.Cluster) (*kubermaticv
...
+		if err = machineClient.ClusterV1alpha1().Machines(metav1.NamespaceSystem).DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{}); err != nil {
+			return nil, fmt.Errorf("failed to delete cluster machines: %v", err)
 		}
+
 		return c, nil
 	}
 
-	err = clusterClient.CoreV1().Nodes().DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{})
+	c, err = cc.updateCluster(c.Name, func(c *kubermaticv1.Cluster) {
+		c.Finalizers = kuberneteshelper.RemoveFinalizer(c.Finalizers, nodeDeletionFinalizer)
```
Comment (alvaroaleman) [inline api/pkg/controller/cluster/deleting.go:97]: This behaviour wasn't introduced in this PR but shouldn't we wait until the machines are _actually_ gone before removing that finalizer as opposed to we _triggered_ their deletion?

Comment (mrIncompetent) [reply api/pkg/controller/cluster/deleting.go:97]: We do: https://github.com/kubermatic/kubermatic/pull/2218/files#diff-50948c993a2ed95209447dad9bdea621R73

File: api/pkg/controller/cluster/deleting.go (hunk 1 above)
Comment (mrIncompetent) [reply api/pkg/controller/cluster/deleting.go]: fixed

File: api/pkg/controller/cluster/deleting.go (hunk 2 above)
Comment (mrIncompetent) [reply api/pkg/controller/cluster/deleting.go]: fixed

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**